lib/fluent/plugin/output.rb

```diff
@@ -326,6 +326,7 @@ def start
           @buffering = prefer_buffered_processing
           if !@buffering && @buffer
             @buffer.terminate # it's not started, so terminate will be enough
+            @buffer = nil
```

Comment:
I see. If so, adding comment is better like "This line is needed for detecting buffer or non-buffer, e.g. used by in_monitor_agent".
Only this line, it looks like for GC.

And we should move this `:buffer` reader from `for test` to above `attr_reader` lines.
https://github.com/fluent/fluentd/blob/24e15f3cfcb46029b1dee080bf50b576f573553e/lib/fluent/plugin/output.rb#L145